### PR TITLE
Fix counting vulnerabilities in listing

### PR DESCRIPTION
**Pull request by ghost**
Replace vuls. counting in listing with simple string "vulnerable".
This increase view rendering. Items are now links to details of scan.

### CODE REVIEW

**Review by ar4s — APPROVED** [review]

File: src/ralph/data_center/admin.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ class DataCenterAssetOperation(OperationViewReadOnlyForExisiting):
 
 
 class DataCenterAssetSecurityInfo(SecurityInfo):
-    url_name = 'datacenter_asset_security_info'
+    url_name = 'security_info'
```
Comment (mkurek) [inline src/ralph/data_center/admin.py]: change it for cluster too: 
```
django.core.urlresolvers.NoReverseMatch: Reverse for 'data_center_cluster_security_info' with arguments '(78698,)' and keyword arguments '{}' not found. 0 pattern(s) tried: []
```

Comment (mkurek) [reply src/ralph/data_center/admin.py]: And, if you change all url_names to `security_info`, breadcrumbs in this tab do not work.

**Review by andrzej-jankowski — APPROVED** [review]

File: src/ralph/security/views.py [hunk 2]
```diff
@@ -17,7 +20,16 @@ def __init__(self, *args, **kwargs):
     def get_queryset(self, request):
         qs = super().get_queryset(request)
         qs = qs.annotate(
-            vulnerabilities_count=Count('securityscan__vulnerabilities')
+            exceeded_vulnerabilities_count=Sum(
```
Comment (mkurek) [inline src/ralph/security/views.py]: this query is sooooo slow - on my instance it runs over 400ms for list of objects and over 3 seconds (!!!) when searching (comparing to 50ms for list and 350ms for search - so queries are now 10x longer). We have to do it better. Could we use `is_patched` here?

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/security/views.py [hunk 3]
```diff
@@ -31,16 +43,19 @@ def scan_status(self, obj):
...
                     html = self._to_span(
                         "alert", "Got vulnerabilities: {}".format(
-                            obj.vulnerabilities_count
+                            obj.exceeded_vulnerabilities_count
                         )
                     )
                 else:
                     html = self._to_span("success", "Host clean")
             else:
                 html = self._to_span("warning", "Scan failed")
```
Comment (mkurek) [inline src/ralph/security/views.py]: where do you find these classes (success, warning) for span? It's not working right now (afair it should be red when host has vulnerabilities)?

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/data_center/admin.py (hunk 1 above)
Comment (ghost) [reply src/ralph/data_center/admin.py]: ok, then i can revert this rename thing.. and create for example some map dict or something, it's productive however could be more errorproof..
what do you think?

Comment (mkurek) [reply src/ralph/data_center/admin.py]: yes. if sth will not be in this map then just don't reverse url (and display it as simple text) and log error.

File: src/ralph/security/views.py (hunk 2 above)
Comment (ghost) [reply src/ralph/security/views.py]: about `is_patched`, I dont see it as solution, could you expand your concept?

Comment (xliiv) [inline src/ralph/security/views.py]: https://github.com/allegro/ralph/blob/cf9ec22565e79334cce588a14103e119edebe8b1/src/ralph/static/src/scss/_styles.scss#L162
from PR:
https://github.com/allegro/ralph/pull/2982/files#diff-3c0e4dc1a693950df4b64a82cfa9b6a2

File: src/ralph/security/views.py (hunk 3 above)
Comment (ghost) [reply src/ralph/security/views.py]: https://github.com/allegro/ralph/blob/cf9ec22565e79334cce588a14103e119edebe8b1/src/ralph/static/src/scss/_styles.scss#L162
from PR:
https://github.com/allegro/ralph/pull/2982/files#diff-3c0e4dc1a693950df4b64a82cfa9b6a2

Comment (mkurek) [reply src/ralph/security/views.py]: ok - so probably i've got some old static files. Ignore this comment.

File: src/ralph/security/views.py [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+from ralph.admin.helpers import get_admin_url
+from ralph.admin.sites import ralph_site
 from ralph.admin.views.extra import RalphDetailView
 from ralph.security.models import Vulnerability
 
 
+logger = logging.getLogger(__name__)
+
+
+def _url_name_for_change_view(obj, view_name):
```
Comment (mkurek) [inline src/ralph/security/views.py]: could we cache it (based on `type(obj)` and `view_name`)? use for example `django.utils.lru_cache`

Comment (mkurek) [reply src/ralph/security/views.py]: and add some docstring here

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/security/views.py [hunk 5]
```diff
@@ -16,7 +17,14 @@
 logger = logging.getLogger(__name__)
 
 
+@lru_cache()
```
Comment (mkurek) [inline src/ralph/security/views.py]: this cache will not work right now - the param to this function has to be type, not object

**Review by mkurek — CHANGES_REQUESTED** [review]

**Review by mkurek — APPROVED** [review]

**Pull request merged**